### ISSUE TITLE
_if issues with empty strings

### ISSUE THREAD

**Issue by david-zacharias** [MathiasPaumgarten/grunt-bake#50]
Hey Mathias,

it's me once again. I have an issue with the `_if` in combination with empty strings in data. When I have a inline section like

```HTML
<!--(bake-start _if="employee.quote")-->
  <p class="quote">{{employee.quote}}</p>
<!--(bake-end)-->
```
and `employee.quote` is an empty string (`""`), the section is still rendered. I tracked down the issue and would say its due to two things:

**1st:** `isFalse( value )` does not consider empty strings to be false. I would say it should. So I recommend add this check to line 129, like:
```JS
if ( value === '' || value === undefined || value === false || string === "false" ) {
```

**2nd:** `validateIf( inlineValues, values )` returns `true` when `value` is in `values`. It does not check if resolved `value` is true or false. I think it is better to check this, so my recommendation is to change line 235 to:
```JS
if ( ! hasValue( value, values ) && isFalse( resolveName( value, values ) ) ) return true;
```

or
```JS
if ( ! hasValue( value, values ) ) return isFalse( resolveName( value, values ) );
```
What do you think?

**Comment by MathiasPaumgarten** [comment]
Those changes sound good to me. Care to create a PR? If not, I'll put it in.

**Comment by david-zacharias** [comment]
Go ahead and put the changes in. There is already a PR with another idea I came up with today. Thanks!

**Comment by MathiasPaumgarten** [comment]
Fixed that in #52
